### DISCUSSION TITLE
how does it deal with guest cache?

### DISCUSSION THREAD

**Discussion by rebbieboi** [spatie/laravel-responsecache#476] [Q&A] [answered]
Since the doc says it only creates a separate cache for logged-in users? so do guests have a shared cache?

**Comment by vinhkhangphung** [comment, marked as answer]
Please take a look at abstract class `BaseCacheProfile`, it implements function `useCacheNameSuffix(Request $request): string` from `CacheProfile`.
```
public function useCacheNameSuffix(Request $request): string
{
    return Auth::check()
        ? (string) Auth::id()
        : '';
}
```

As you can see, it returns either User ID or empty string, that is the string use to differ between a logged-in users and guest users. Moreover, you can return as many different string due to as many conditions as you need. If you want both of them to use the same cache, just simply return empty string from the function.
